From 6638f2ed40e9375b3dd9e72fcac2f726ee218448 Mon Sep 17 00:00:00 2001
From: Ryan McLean <rgmclean@deloitte.co.uk>
Date: Mon, 19 Aug 2024 06:39:46 +0100
Subject: [PATCH] Use Include list

---
 action.yml | 4 ++++
 index.ts   | 8 ++++++++
 2 files changed, 12 insertions(+)

diff --git a/action.yml b/action.yml
index e2a7dc1..78cf1a1 100644
--- a/action.yml
+++ b/action.yml
@@ -33,6 +33,9 @@ inputs:
     description: 'Name of the release to copy'
     required: true
     default: ''
+  include-assets:
+    description: 'List of assets to copy'
+    required: true
 outputs: {}
 runs:
   using: 'docker'
@@ -45,6 +48,7 @@ runs:
     DEST_OWNER: "${{ inputs.dest-owner }}"
     DEST_REPO: "${{ inputs.dest-repo }}"
     TEMP_DIR: "${{ inputs.temp-dir }}"
+    INCLUDE_ASSETS: "${{ inputs.include-assets}}"
     BODY_REPLACE_REGEX: "${{ inputs.body-replace-regex }}"
     BODY_REPLACE_WITH: "${{ inputs.body-replace-with }}"
   args: [ "${{ inputs.release-name }}" ]
diff --git a/index.ts b/index.ts
index 81304f3..294e798 100644
--- a/index.ts
+++ b/index.ts
@@ -13,6 +13,7 @@ const {
     DEST_OWNER,
     DEST_REPO,
     TEMP_DIR,
+    INCLUDE_ASSETS,
     BODY_REPLACE_REGEX,
     BODY_REPLACE_WITH,
 } = process.env;
@@ -62,6 +63,7 @@ async function downloadAsset(
 
 const downloadAssets = async (
     apiKey: string,
+    includeAssets: string[],
     owner: string,
     repo: string,
     tag: string,
@@ -87,6 +89,9 @@ const downloadAssets = async (
     console.log(`Release has ${assets.data.length} assets`);
 
     for (const asset of assets.data) {
+        if (!includeAssets.includes(asset)) {
+            continue
+        }
         const assetResponse = await fetchAssetDetails(connection, owner, repo, asset.id);
         const fileName = assetResponse.name;
 
@@ -153,10 +158,13 @@ const copyRelease = async () => {
         fs.mkdirSync(tempDir);
     }
 
+    const includeAssets: string[] = INCLUDE_ASSETS.split(" ");
+
     const sourceOwner = SOURCE_OWNER!;
     const sourceRepo = SOURCE_REPO!;
     const release = await downloadAssets(
         SOURCE_API_KEY!,
+        includeAssets,
         sourceOwner,
         sourceRepo,
         releaseTag,
-- 
2.44.0

